File: backend/tour-design-api/src/main/java/com/tourdesign/platform/model/ClientModel.java
package com.tourdesign.platform.model;

import lombok.*;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ClientModel {
    private Long id;

    private String firstName;
    private String lastName;
    private LocalDate birthDate;
    private String gender;
    private String nationality;
    private String email;
    private String phone;
    private Integer numberOfChildren;
    private String occupation;
    private Double annualIncome;
    private String languages;
    private LocalDateTime registrationDate;
    private Boolean active;

    private List<ClientPreferenceModel> preferences;
    private List<RecommendationHistoryModel> recommendationHistory;
}

File: frontend/tour-design-web/src/app/models/client.model.ts
// src/app/models/client.model.ts
import { ClientPreferenceModel } from './client-preference.model';
import { RecommendationHistoryModel } from './recommendation-history.model';

export interface ClientModel {
  id: number;
  firstName: string;
  lastName: string;
  birthDate?: string;
  gender?: string;
  nationality?: string;
  email?: string;
  phone?: string;
  numberOfChildren?: number;
  occupation?: string;
  annualIncome?: number;
  languages?: string;
  registrationDate?: string;
  active?: boolean;

  preferences?: ClientPreferenceModel[];
  recommendationHistory?: RecommendationHistoryModel[];
}
